lib/msal-common/src/response/ResponseHandler.ts

```diff
@@ -133,16 +133,37 @@ export class ResponseHandler {
     /**
      * Function which validates server authorization token response.
      * @param serverResponse
+     * @param accessTokenRefresh
      */
     validateTokenResponse(
-        serverResponse: ServerAuthorizationTokenResponse
+        serverResponse: ServerAuthorizationTokenResponse,
+        accessTokenRefresh?: boolean
     ): void {
         // Check for error
         if (
             serverResponse.error ||
             serverResponse.error_description ||
             serverResponse.suberror
         ) {
+
+            // check if 500 error
+            if (accessTokenRefresh && serverResponse.status && String(serverResponse.status)[0] === "5") {
```

Comment:
Magic string - please put the possible serverResponse statuses in an enum

Reply:
+1 

Also, why check the first character only?

Reply:
@Robbie-Microsoft wants to check that the error code is between 500 and 599, which indicates there's something wrong with the STS.

Reply:
Can we just create a constant then, which says, `ServerErrorRangeHint` or something?

Reply:
Agreed, int comparison would be better than converting to string in this case